tests/sentry/api/endpoints/test_project_codeowners.py

```diff
@@ -405,172 +405,41 @@ def test_post_with_streamline_targeting(self, get_codeowner_mock_file):
-            user=self.member_user_delete, external_name="@delete", integration=self.integration
-        )
-        self.data["raw"] = "docs/*  @delete"
-
-        # Post without the streamline-targeting-context flag
-        with self.feature({"organizations:integrations-codeowners": True}):
-            self.client.post(self.url, self.data)
-
-            # Test get after with the streamline-targeting-context flag
-            with self.feature({"organizations:streamline-targeting-context": True}):
-                self.external_delete_user.delete()
-                response = self.client.get(self.url)
-                assert response.data[0]["schema"] == {"$version": 1, "rules": []}
-
-    @patch(
-        "sentry.integrations.mixins.repositories.RepositoryMixin.get_codeowner_file",
-        return_value={"html_url": "https://github.com/test/CODEOWNERS"},
-    )
-    def test_get_multiple_rules_deleted_owners_with_streamline_targeting(
-        self, get_codeowner_mock_file
-    ):
-        self.member_user_delete = self.create_user("member_delete@localhost", is_superuser=False)
-        self.create_member(
-            user=self.member_user_delete,
-            organization=self.organization,
-            role="member",
-            teams=[self.team],
-        )
-        self.external_delete_user = self.create_external_user(
-            user=self.member_user_delete, external_name="@delete", integration=self.integration
-        )
-        self.member_user_delete2 = self.create_user("member_delete2@localhost", is_superuser=False)
-        self.create_member(
-            user=self.member_user_delete2,
-            organization=self.organization,
-            role="member",
-            teams=[self.team],
-        )
-        self.external_delete_user2 = self.create_external_user(
-            user=self.member_user_delete, external_name="@delete2", integration=self.integration
-        )
-        self.data[
-            "raw"
-        ] = "docs/*  @delete\n*.py @getsentry/ecosystem @delete\n*.css @delete2\n*.rb @NisanthanNanthakumar"
-
-        # Post without the streamline-targeting-context flag
-        with self.feature({"organizations:integrations-codeowners": True}):
-            self.client.post(self.url, self.data)
-
-            # Test get after with the streamline-targeting-context flag
-            with self.feature({"organizations:streamline-targeting-context": True}):
-                self.external_delete_user.delete()
-                self.external_delete_user2.delete()
-                response = self.client.get(self.url)
-                assert response.data[0]["schema"] == {
-                    "$version": 1,
-                    "rules": [
-                        {
-                            "matcher": {"type": "codeowners", "pattern": "*.py"},
-                            "owners": [{"type": "team", "name": "tiger-team", "id": self.team.id}],
-                        },
```

Comment:
Shouldn't we be keeping these tests? They seem to test this functionality

Reply:
I chatted with @jangjodi about these tests, and I believe they were tests for the interim while the feature flag was being rolled out, making sure that the legacy behaviour was still preserved. With the flag GA'd now, my assumption is that these tests are no longer valid. If that's not the case I can totally rewrite them rather than delete them, but I wanna make sure that 'deleted owner' behaviour is something we intend to keep.

Reply:
After a chat with jodi, it looks like we can keep these tests, I'll refactor them to make sure they pass :+1:

Reply:
Sounds good, just wanted to make sure we don't lose coverage. If there's duplication, feel free to consolidate as well

Reply:
Okay, resolved in 1146d752b2, it was a bug. `add_owner_id_to_schema`were only added if `id` wasn't found in the schema, but also as a side-effect, we were removing deleted owners. This changes it to always run the check to delete from the schema by always calling `create_schema_from_issue_owners`. lmk what you think!